src/os/bluestore/BlueStore_debug.cc

```diff
@@ -0,0 +1,247 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2023 IBM
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+#include <cstdint>
+#include <ostream>
+#include "BlueStore.h"
+
+static const std::string transition_table[26] = {
+"bcdfghjklmnprstuvxyz", //a
+"aeiloruy",//b
+"aeiloruvy",//c
+"aeilmnoruvy",//d
+"bcdfghjklmnprstvxz",//e
+
+"ailou",//f
+"aeilnoru",//g
+"aeiloru",//h
+"dfghklmnpqrstvwx",//i
+"aeiou",//j
+
+"aeiloru",//k
+"aeimnou",//l
+"aeinotuy",//m
+"aeiou",//n
+"bcdfghjklmnpqrstvwxz",//o
+"aehiloruy",//p
+
+"aeiloru",//q
+"adefiklmnotuvy",//r
+"aehiklmnopqrtuvwy",//s
+"acefhiklmnorsuvwy",//t
+"bcdfghklmnpqrsvwxyz",//u
+
+"acdeiklmorsu",//v
+"aehilnorstu",//w
+"aeilnorstuy",//x
+"aehinorsuxz",//y
+"aeiouy" //z
+};
+
+std::string int_to_fancy_name(uint64_t x)
+{
+  std::string result;
+  uint8_t c = x % 26;
+  x = x / 26;
+  result.push_back(c+'a');
+  while (x > 0) {
+    uint8_t range = transition_table[c].length();
+    uint8_t p = x % range;
+    c = transition_table[c][p] - 'a';
+    x = x / range;
+    result.push_back(c+'a');
+  }
+  return result;
+}
+
+// Use special printing for multiplies of 1024; print K suffix
+struct maybe_K {
+  uint32_t x;
+  maybe_K(uint32_t x) : x(x) {}
+};
+std::ostream &operator<<(std::ostream &out, const maybe_K &k) {
+  if (((k.x & 0x3ff) == 0) && (k.x != 0)) {
+    if (k.x != 0x400)
+      out << (k.x / 0x400);
+    out << "K";
+  } else {
+    out << std::hex << k.x << std::dec;
+  }
+  return out;
+}
+
+std::ostream& operator<<(std::ostream& out, const BlueStore::Buffer& b);
+std::ostream& operator<<(std::ostream& out, const BlueStore::Blob::printer &p)
+{
+  using P = BlueStore::printer;
+  out << "Blob(";
+  if (p.mode & P::ptr) {
```

Comment:
Definitely I'd prefer flag names being defined in upper case

Reply:
Done.